ibm/service/codeengine/resource_ibm_code_engine_build.go

```diff
@@ -0,0 +1,586 @@
+				Description:  "Commit, tag, or branch in the source repository to pull. This field is optional if the `source_type` is `git` and uses the HEAD of default branch if not specified. If the `source_type` value is `local`, this field must be omitted.",
+			},
+			"source_secret": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_build", "source_secret"),
+				Description:  "Name of the secret that is used access the repository source. This field is optional if the `source_type` is `git`. Additionally, if the `source_url` points to a repository that requires authentication, the build will be created but cannot access any source code, until this property is provided, too. If the `source_type` value is `local`, this field must be omitted.",
+			},
+			"source_type": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "git",
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_build", "source_type"),
+				Description:  "Specifies the type of source to determine if your build source is in a repository or based on local source code.* local - For builds from local source code.* git - For builds from git version controlled source code.",
+			},
+			"strategy_size": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "medium",
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_build", "strategy_size"),
+				Description:  "Optional size for the build, which determines the amount of resources used. Build sizes are `small`, `medium`, `large`, `xlarge`.",
+			},
+			"strategy_spec_file": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "Dockerfile",
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_build", "strategy_spec_file"),
+				Description:  "Optional path to the specification file that is used for build strategies for building an image.",
+			},
+			"timeout": &schema.Schema{
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      600,
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_build", "timeout"),
+				Description:  "The maximum amount of time, in seconds, that can pass before the build must succeed or fail.",
+			},
+			"created_at": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date when the resource was created.",
+			},
+			"entity_tag": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The version of the build instance, which is used to achieve optimistic locking.",
+			},
+			"href": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "When you provision a new build,  a URL is created identifying the location of the instance.",
+			},
+			"id": &schema.Schema{
```

Comment:
kindly change id to build_id
Make respective changes in all the references of this field in this resource and in docs

Reply:
Changed the ID according to suggestion